update-service-brokers-docs (#913)
meshcloud/meshcloud-docs: docs/marketplace.vs-building-blocks.md

|                                           | Building Blocks                                                                   | Service Brokers                                                                                                                       |
|-------------------------------------------|-----------------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------|
| Provision resources with Terraform        | ✅ This works very easily                                                          | 🟨 Works using [UniPipe Service Broker](https://github.com/meshcloud/unipipe-service-broker), but requires more implementation effort |

> | Run Custom Processes outside of Terraform | ✅ Async provisioning is supported and allows writing back status updates from external systems into meshStack                                                                                 | ✅ Service Brokers run on OSB API which means you can build anything you want as long as the interface is compliant with OSB API       |

Reviewer: Shouldn't we rather mention that you can trigger external processes using Terraform? Like committing a file into a CI/CD repo

Reviewer 2: That the triggering is to be done with terraform is correct. The important information from my perspective is not the specific form of the glue you need, but that you can do async provisioning via tools you control and that this isn't an afterthought but something we designed meshStack for

| Automated Deletion of Resource            | ✅ Terraform Building Blocks can run `tf destroy` automatically                    | 🟨 Possible but logic needs to be implemented in the service broker                                                                   |

> | Request User Configuration                | ✅ Further variable types such as "select" are planned                             | ✅ Works with JSON schema and allows for a lot of different variations                                                                 |

Reviewer: ```suggestion
| Request User Configuration                | ✅ You can define a form for your users via the UI                           | ✅ Works with JSON schema and allows for a lot of different variations                                                                 |
```

| Manage Instances via API                  | ✅                                                                                 | ✅ But missing deletion of meshServiceInstances                                                                                        |
| Manage Definitions via API                | ⏰                                                                                 | ❌                                                                                                                                     |
| Custom Pricing Models                     | ✅ via [meshResourceUsageReports API](https://docs.meshcloud.io/billing-api/index.html#_put_meshresourceusagereports)                                                                                | ✅ Can be made part of the definition                                                                                          |

> | Sandbox Development Process               | ⏰ After creating a new Building Block, it is directly available to all workspaces | ✅ Developers can create a private service broker that cannot be booked by any other workspaces                                        |

Reviewer: ```suggestion
| Sandbox Development Process               | ✅ Developers can test drafts of building blocks that cannot be booked by any other workspaces | ✅ Developers can create a private service broker that cannot be booked by any other workspaces                                        |
```

| Versioning                                | ✅                                   | ❌ Not possible. There is no such thing as service versions                                                                            |
| Works without cloud platforms             | ✅ Building Blocks can create a tenant as part of a custom platform                                 | ✅ Service Brokers work without having a cloud tenant                                                                                  |
| View all consumption in Admin Area        | ✅ There is a list with all building blocks in the Admin Area                      | ❌ There is only a list of service brokers but not service instances                                                                   |
| Tag and Policy Support                    | ✅                                                                                  | ❌                                                                                                                                     |

> | Integration on Landing Zones              | ✅ Can be set as recommended or mandatory on a landing zone                        | 🟨 For tenant-aware services                                                                                                          |

Reviewer: ```suggestion
| Integration on Landing Zones              | ✅ Can be set as recommended or mandatory on a landing zone                        | ❌                                                                                                           |
```
We (wanted to) deprecate this so let's not mention it anymore.
